src/TestFramework/Factory.php

```diff
@@ -105,19 +107,27 @@ public function create(string $adapterName, bool $skipCoverage): TestFrameworkAd
             );
         }
 
-        if ($adapterName === TestFrameworkTypes::CODECEPTION) {
-            $codeceptionConfigPath = $this->configLocator->locate(TestFrameworkTypes::CODECEPTION);
+        foreach (GeneratedExtensionsConfig::EXTENSIONS as $packageName => $installedExtension) {
```

Comment:
Would it be worthwhile to inject `GeneratedExtensionsConfig::EXTENSIONS` into the factory? That would allow us to test it.

Reply:
Well, this unfortunately requires to rewrite `TestFrameworkFinder` and inject it as well, because

* it's created inside factory like `new TestFrameworkFinder(...)`
* test framework name is passed to its constructor so it's not reusable for different adapters -> we need to rewrite `TestFrameworkFinder` to probably pass adapter name to `find()` rather than to constructor

I'm not sure I want to do it in this PR.. What do you think?

Reply:
I'm not sure i follow. The factory is created from inside the container, so we should be able to inject the extension into the constructor there if i'm not mistaken.

Reply:
I did it locally, but tests failing because of

```
Infection\FileSystem\Finder\Exception\FinderException: Unable to locate a dummy executable on local system. Ensure that dummy is installed and available.
```

Look at this line:

https://github.com/infection/infection/blob/b70eb6b88fe362797bf2ecaa2326a89d8dcfdc64/src/TestFramework/Factory.php#L113

`TestFrameworkFinder` is hardcoded and also should be injected to be able to mock it. And since it gets test framework parameter to *constructor* rather than to `find()` method. - we can't mock it correctly for all test framework. This class IMO should be completely rewritten as I mentioned in the prev comment